src/prefect/engine/cloud/flow_runner.py

```diff
@@ -72,21 +72,21 @@ def __init__(self, flow: Flow, state_handlers: Iterable[Callable] = None) -> Non
     def _heartbeat(self) -> bool:
         try:
             # use empty string for testing purposes
-            flow_run_id = prefect.context.get("flow_run_id", "")  # type: str
-            self.client.update_flow_run_heartbeat(flow_run_id)
+            self.flow_run_id = prefect.context.get("flow_run_id", "")  # type: str
```

Comment:
+1

Reply:
I wanted to use this interface to get everyone's input on where we want the boundaries for this PR to be. The heartbeat machinery already mutates this class state to present information that contains the `flow_run_id`, previously this took the form of exposing `self.heartbeat_cmd`. I'm not attached to this pattern at all, but in stopping here I had two thoughts: trying to completely remove all of the old heartbeat idiom touches quite a bit of code, as it's tested for several places, and for this stab at drawing a reasonable stopping point, it was to use a similar mechanism to pass information to the new heartbeat (mutating the state once `_heartbeat` is called)--which I do not love. I was going to make another PR against this one just to sketch out what it would look like to go all the way, but if everyone dislikes this pattern as much as I do maybe we should just, at the very least, take this piece out.